### ISSUE TITLE
Pre-compute bucket-specific outputs in lookup table

### ISSUE THREAD

**Issue by zhixiangteoh** [neel-one/nocap#3]
Need a way to compute function outputs corresponding to (perhaps the median of) each input bucket after determining bucket splits, regardless of the outputs that the user program generates.

For example, given a program that computes these values for `log10`:
|**input**|**output**|
|-|-|
|1|0|
|1.5|0.17609...|
|5|0.69897...|
|100|2|

NOCAP currently only "sees" and assigns the program's generated output values to buckets that don't have a program-generated output, via this line:

```python
for i in range(len(tb)):
    if tb[i] is None:
        tb[i] = tb[i-1]
```

Instead, we want a lookup table that has outputs computed for each bucket, like so (for `log10`):
|**bucket**|**value**|
|-|-|
|1-5|`log10(2.5)`|
|6-10|`log10(7.5)`|
|...|...|
|96-100|`log10(97.5)`|

**Comment by zhixiangteoh** [comment]
Assuming only functions in `math.h`, I can think of two ways, one "hacky":

1. [Hacky] Compute directly in Python script, via the Python version of the math.h function
2. Spawn shell subprocess to compile and run "proxy" C program with code like `tb[bucket] = log10(bucket_median)`, and prints the whole table to a file, which we then parse into a hardcoded C program similar to the current `nocap_exp.c`, perhaps via another shell subprocess.

Thoughts @neel-one?

**Comment by neel-one** [comment]
I'm not sure we should care about bucket median vs bucket min vs bucket max. Ideally, we want to force the buckets to be small enough such that any input in the interval should be okay to leave as the table value. That is asking a lot from the profile data, though.

Either way, I'd rather have accuracy be dependent on profile data rather than us optimizing something like bucket median/mean/etc.

Agree or disagree?

**Comment by zhixiangteoh** [comment]
Fixed by #4
